File: lib/domain/shared/validation_helpers.rb
# frozen_string_literal: true

module Domain
  module Shared
    module ValidationHelpers
      # ActiveSupportのblank?とpresent?のRails非依存代替実装
      # 既存のblank?/present?と同じ振る舞いを維持

      def self.present?(value)
        !blank?(value)
      end

      def self.blank?(value)
        case value
        when nil, false
          true
        when true
          false
        when String
          value.strip.empty?
        when Array, Hash
          value.empty?
        else
          false
        end
      end

      # Array() のRails非依存代替実装
      def self.to_array(value)
        case value
        when nil
          []
        when Array
          value
        else
          [value]
        end
      end
    end
  end
end

File: crates/agrr-domain/src/shared/validation/validation_helpers.rs
//! Ruby: `Domain::Shared::ValidationHelpers`

use serde_json::Value;

use crate::shared::hash::to_array;

/// Ruby: `ValidationHelpers.blank?`
pub fn blank(value: &Value) -> bool {
    crate::shared::hash::blank(value)
}

/// Ruby: `ValidationHelpers.present?`
pub fn present(value: &Value) -> bool {
    crate::shared::hash::present(value)
}

/// Ruby: `ValidationHelpers.to_array`
pub fn to_array_value(value: Option<&Value>) -> Vec<Value> {
    to_array(value)
}

#[cfg(test)]
mod validation_validation_helpers_test_inline {
    use super::*;
    include!(concat!(env!("CARGO_MANIFEST_DIR"), "/test/shared/validation_validation_helpers_test.rs"));
}
